src/os/bluestore/BlueStore.cc

```diff
@@ -6445,7 +6474,9 @@ void BlueStore::_txc_state_proc(TransContext *txc)
       }
       txc->shared_blobs_written.clear();
       if (cct->_conf->bluestore_sync_submit_transaction &&
-	  fm->supports_parallel_transactions()) {
+	  fm->supports_parallel_transactions() &&
+          (1 == kv_sync_thread_list.size())) {
```

Comment:
I guess the whole purpose of  this block was to take load away from single kv_sync thread.
Now, if there are multiple kv_sync threads there is no point of separating out txcs like that as this part is still single threaded in case of 4k min_alloc_size.